drivers/mbox/mbox_sophgo_cvi.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2024 honglin leng <a909204013@gmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/mbox.h>
+#include <zephyr/irq.h>
+#include <string.h>
+
+#define DT_DRV_COMPAT sophgo_cvi_mailbox
+
+#define GET_CPUx(base, cpu_num) (base + 0x10 + (0x10 * (cpu_num)))
+
+#define MBOX_INT_CLR_OFFSET    0x00
+#define MBOX_INT_DONE_OFFSET   0x08
+#define MBOX_INT_TRIGER_OFFSET 0x60
+#define MBOX_BUFFER_OFFSET     0x400
+
+#define CPUx_INT_DONE(base, cpu)   (GET_CPUx(base, cpu) + MBOX_INT_DONE_OFFSET)
+#define CPUx_INT_CLEAR(base, cpu)  (GET_CPUx(base, cpu) + MBOX_INT_CLR_OFFSET)
+#define CPUx_INT_TRIGER(base, cpu) (base + MBOX_INT_TRIGER_OFFSET)
+#define CPUx_BUFFER(base)          (base + MBOX_BUFFER_OFFSET)
+
+#define CPUx_INT_ENABLE(base, cpu) (base + (cpu * 0x04))
+
+#define MAILBOX_MAX_NUM 8
+
+struct mbox_cvi_data {
+	mbox_callback_t cb[MAILBOX_MAX_NUM];
+	void *user_data[MAILBOX_MAX_NUM];
+	const struct device *dev;
+};
+
+static struct mbox_cvi_data cvi_mbox_data;
+static struct mbox_cvi_conf {
+	mm_reg_t base;
+	mm_reg_t mbox_buffer;
+	uint32_t rx_cpu;
+	uint32_t tx_cpu;
+} cvi_mbox_conf = {
+	.base = DT_INST_REG_ADDR(0),
+	.mbox_buffer = DT_INST_REG_ADDR(0) + MBOX_BUFFER_OFFSET,
+	.tx_cpu = DT_INST_PROP(0, tx_cpu),
+	.rx_cpu = DT_INST_PROP(0, rx_cpu),
+};
+
+static void mbox_isr(const struct device *dev)
+{
+	uint8_t set_val;
+	uint8_t valid_val;
+	uint8_t tmp_valid_val;
+	struct mbox_msg msg;
+	struct mbox_cvi_data *data = dev->data;
+	const struct mbox_cvi_conf *conf = dev->config;
+
+	set_val = sys_read8(CPUx_INT_DONE(conf->base, conf->rx_cpu));
+	if (set_val) {
+		for (int i = 0; i < MAILBOX_MAX_NUM; i++) {
+			valid_val = set_val & (1 << i);
+			if (valid_val) {
+				msg.data = (unsigned long *)conf->mbox_buffer + i;
+				sys_write8(valid_val, CPUx_INT_CLEAR(conf->base, conf->rx_cpu));
+				tmp_valid_val =
+					sys_read8(CPUx_INT_ENABLE(conf->base, conf->rx_cpu));
+				tmp_valid_val &= ~valid_val;
+				sys_write8(tmp_valid_val,
+					   CPUx_INT_ENABLE(conf->base, conf->rx_cpu));
+				if (data->cb[i] != NULL) {
+					data->cb[i](dev, i, data->user_data[i], &msg);
+					*(unsigned long *)msg.data = 0x0;
+				}
+			}
+		}
+	}
+}
+
+static int mbox_cvi_send(const struct device *dev, uint32_t channel, const struct mbox_msg *msg)
+{
+	uint8_t tmp_mbox_info;
+	const struct mbox_cvi_conf *conf = dev->config;
```

Comment:
suggestion: if not multi-instance, you can also access static config/data directly, as they're not device-dependent.